client/src/routes/Root.tsx

```diff
@@ -17,6 +20,37 @@ export default function Root() {
   const fileMap = useFileMap({ isAuthenticated });
   const assistantsMap = useAssistantsMap({ isAuthenticated });
 
+  const [showTerms, setShowTerms] = useState(false);
+  const { data: config } = useGetStartupConfig();
+
+  useEffect(() => {
+    if (isAuthenticated && config?.interface?.termsOfService?.modalAcceptance) {
+      fetch('/api/user/terms', {
```

Comment:
don't make raw requests, use react query (in data-provider). This apply for both the post and get request